src/common/address.cpp

```diff
@@ -105,7 +105,8 @@ Error Address::ToString(std::string &aAddr) const
 Address Address::FromString(const std::string &aAddr)
 {
     Address ret;
-    ret.Set(aAddr);
+    auto    error = ret.Set(aAddr);
+    ASSERT(error == Error::kNone);
```

Comment:
Is `assert` too strict for parsing error?

Reply:
- The semantic of `FromString` is `don't accept invalid aAddr`.
- Each usage of `FromString` is passed a string literal, So we are rejecting programming bug.
- FromString is internally accessible.